cmd/tools/testwick/testwicker_test.go

```diff
@@ -0,0 +1,734 @@
+				wicker.userID = "1234"
+				wicker.installation = &model.InstallationDTO{
+					Installation: &cmodel.Installation{
+						ID: "123",
+					},
+				}
+				mmRequester.EXPECT().
+					CreateTeam(gomock.Any()).Return(nil, &mmodel.Response{
+					StatusCode: 400,
+					Error:      &mmodel.AppError{Message: "any mattermost API error"},
+				})
+			},
+			expected: func(wicker *TestWicker, err error) {
+				assert.NotNil(t, err)
+				assert.Contains(t, err.Error(), "failed to create a team")
+			},
+		},
+		{
+			description: "create team success",
+			setup: func(wicker *TestWicker) {
+				wicker.userID = "1234"
+				wicker.installation = &model.InstallationDTO{
+					Installation: &cmodel.Installation{
+						ID: "123",
+					},
+				}
+				mmRequester.EXPECT().
+					CreateTeam(gomock.Any()).Return(&mmodel.Team{
+					Id: "1234",
+				}, &mmodel.Response{
+					StatusCode: 201,
+				})
+			},
+			expected: func(wicker *TestWicker, err error) {
+				assert.Nil(t, err)
+				assert.NotEmpty(t, wicker.teamID)
+			},
+		},
+	}
+
+	for _, v := range samples {
+		t.Run(v.description, func(t *testing.T) {
+			wicker.installation = &model.InstallationDTO{
+				Installation: &cmodel.Installation{
+					ID: "123",
+				},
+			}
+			v.setup(wicker)
+
+			f := wicker.CreateTeam()
+			err := f(wicker, context.TODO())
+
+			v.expected(wicker, err)
+		})
+	}
+}
+
+func TestAddTeamMember(t *testing.T) {
+	logger := testlib.MakeLogger(t)
+	gmctrl := gomock.NewController(t)
+	mmRequester := mocks.NewMockMattermostRequester(gmctrl)
+	provisionerRequester := mocks.NewMockProvisionerRequester(gmctrl)
+	wicker := NewTestWicker(provisionerRequester, mmRequester, logger)
+
+	samples := []struct {
+		description string
+		setup       func(wicker *TestWicker)
+		expected    func(wicker *TestWicker, err error)
+	}{
+		{
+			description: "team ID is missing",
+			setup:       func(wicker *TestWicker) {},
+			expected: func(wicker *TestWicker, err error) {
+				assert.NotNil(t, err)
+				assert.Contains(t, err.Error(), "You need to create a team and a user first")
+			},
+		},
+		{
+			description: "user ID is missing",
+			setup:       func(wicker *TestWicker) {},
+			expected: func(wicker *TestWicker, err error) {
+				wicker.teamID = "1234"
+				assert.NotNil(t, err)
+				assert.Contains(t, err.Error(), "You need to create a team and a user first")
+			},
+		},
+		{
+			description: "add api team member error",
+			setup: func(wicker *TestWicker) {
+				wicker.teamID = "1234"
+				mmRequester.EXPECT().
+					AddTeamMember(gomock.Any(), gomock.Any()).Return(nil, &mmodel.Response{
+					StatusCode: 400,
+					Error:      &mmodel.AppError{Message: "any mattermost API error"},
+				})
+			},
+			expected: func(wicker *TestWicker, err error) {
+				assert.NotNil(t, err)
+				assert.Contains(t, err.Error(), "failed to add a team member")
+			},
+		},
+		{
+			description: "add team member success",
+			setup: func(wicker *TestWicker) {
+				wicker.userID = "1234"
```

Comment:
This test is failing on my machine, I think we need to specify `teamID` also.
```suggestion
				wicker.teamID = "1234"
				wicker.userID = "1234"
```

Reply:
Sorry I think mocks need to be adjusted too.

Reply:
let me check